pkg/cluster/deploybaseresources.go

```diff
@@ -35,6 +39,47 @@ func (m *manager) createDNS(ctx context.Context) error {
 	return m.dns.Create(ctx, m.doc.OpenShiftCluster)
 }
 
+func (m *manager) createOIDC(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil || m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+
+	blobContainerName := env.OIDCBlobContainerPrefix + m.doc.ID
```

Comment:
Few days back we found out that $web container under storage account will let us access data keeping anonymous blob access disabled when you enable static website. This should comply with the ongoing MSFT security wave and fit with our need. I would suggest using $web container here. Here is the more [details on how this works](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website#impact-of-setting-the-access-level-on-the-web-container) and [how to setup this.](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website-host). You can also look at the [OIDC infra-PR to see how this is done on INT and Prod](https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/ARO-Pipelines/pullrequest/10275068)
cc - @weinong

Reply:
I'm adding a hold label here while we address this comment.

Reply:
Resolving this comment thread since Rajdeep has committed these changes.